ckanext/datagov_inventory/plugin.py

```diff
@@ -9,33 +9,23 @@
 import logging
 import re
 
-
 log = logging.getLogger(__name__)
 
-# starting auth functions.
-# Most auth functions should only allow logged in users to access.
-
-
-# Implemented as a decorator
-def datagov_disallow_anonymous_access(func=None):
-    @toolkit.chained_auth_function
-    @toolkit.auth_disallow_anonymous_access
-    def _wrapper(next_auth, context, dict_data=None):
-        if func:
-            # We're called in the decorator context, call the function
-            return func(next_auth, context, dict_data)
-        else:
-            # We were called as a function, just call the next_auth function
-            return next_auth(context, dict_data)
-
-    # We're always applying the auth_disallow_anonymous_access decorator.
-    return _wrapper
 
+@toolkit.auth_allow_anonymous_access
+@toolkit.chained_auth_function
+def restrict_anon_access(next_auth, context, data_dict):
+    if context["user"]:
+        return next_auth(context, data_dict)
+    else:
+        return {'success': False}
```

Comment:
If the other test passes, I think it's safe to merge.  I think this new solution is interesting, but if it works, I guess it's fine.  My only concern would be if there is no `user` key in the `context` dictionary for any reason.

Reply:
i can pair and show you the actual thread, but the gist of the response from the CKAN gods is that:

> Using current_user in auth functions is ok, but it bounds the function to the request context too much. Basically, you function can work only inside flask's request context and it always checks the current user.

>it gets pretty complex, so CKAN recommends a different strategy
look at the context argument of the auth function. It's a special dictionary, that always have a user key inside it. This key have one of the following values, depending on the way the auth function is called:

>if it's called by CKAN, it will contain either "" during the anonymous request, or username(as string) during authenticated request.